app/src/main/java/ch/epfl/culturequest/ui/scan/ScanFragment.java

```diff
@@ -84,8 +93,12 @@ public void onSurfaceTextureUpdated(@NonNull SurfaceTexture surfaceTexture) {
                                         String serializedArtDescription = DescriptionSerializer.serialize(artDescription);
                                         intent.putExtra("artDescription", serializedArtDescription);
                                         intent.putExtra("imageUri", lastlyStoredImageUri.toString());
-                                        loadingAnimation.stopLoading();
                                         startActivity(intent);
+
+                                        // Reset state of the scan fragment
+                                        loadingAnimation.stopLoading();
+                                        scanningLayout.setVisibility(View.INVISIBLE);
+                                        currentProcessing = null;
                                     });
```

Comment:
Where have you fixed the problem of scanning several times ?

Reply:
I added a view to make a darker background for the loading. And this view is made clickable so that it catches the click event before the scan button underneath or the help button.

Reply:
oh ok perfect, that's smart !